DESIGN.md

```diff
@@ -0,0 +1,47 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request. The underlying default
+[transport](https://pkg.go.dev/net/http#Transport) uses [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) for
+HTTP URLs and either [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) or 
+[HTTP/2.0](https://datatracker.ietf.org/doc/html/rfc7540) for HTTPS URLs depending on whether the server supports
+[HTTP/2.0](https://datatracker.ietf.org/doc/html/rfc7540). Non-standard protocols (e.g., 
+[SPDY](https://tools.ietf.org/id/draft-ietf-httpbis-http2-00.html), 
+[QUIC](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34)) are not supported.
+* Support usage of requests to URLs directed to either `http` or `https`. The current version of this provider is built
+with [Go 1.17](https://go.dev/doc/go1.17) which [supports](https://go.dev/doc/go1.17#minor_library_changes) 
+[TLS/1.0](https://www.ietf.org/rfc/rfc2246.txt) ([deprecated](https://datatracker.ietf.org/doc/rfc8996/)), 
+[TLS/1.1](https://datatracker.ietf.org/doc/html/rfc4346) ([deprecated](https://datatracker.ietf.org/doc/rfc8996/)), 
+[TLS/1.2](https://datatracker.ietf.org/doc/html/rfc5246) and 
+[TLS/1.3](https://datatracker.ietf.org/doc/html/rfc8446) (supported since [Go 1.13](https://go.dev/doc/go1.13#tls_1_3)). 
```

Comment:
I think we can omit the final parenthesis statement here since as saying Go 1.13 won't mean much for practitioners (as they'd have to determine which provider versions were built and released on that Go version) 👍

Reply:
Good point, have removed.